gptme/tools/save.py

```diff
@@ -5,64 +5,86 @@
 from collections.abc import Generator
 from pathlib import Path
 
+from .base import Parameter
+
 from ..message import Message
 from ..util import print_preview
 from .base import ConfirmFunc, ToolSpec, ToolUse
 from .patch import Patch
 
-# FIXME: this is markdown-specific instructions, thus will confuse the XML mode
 instructions = """
-To write to a file, use a code block with the language tag: `save <path>`
+Create or overwrite a file with the given content.
 
 The path can be relative to the current directory, or absolute.
 If the current directory changes, the path will be relative to the new directory.
 """.strip()
 
+instructions_format = {
+    "markdown": "To write to a file, use a code block with the language tag: `save <path>`",
+}
+
 instructions_append = """
-To append to a file, use a code block with the language tag: `append <path>`
+Append the given content to a file.`.
 """.strip()
 
-examples = f"""
+instructions_format_append = {
+    "markdown": """
+Use a code block with the language tag: `append <path>` 
+to append the code block content to the file at the given path.""".strip(),
+}
+
+
+def examples(tool_type):
+    return f"""
 > User: write a hello world script to hello.py
-{ToolUse("save", ["hello.py"], 'print("Hello world")').to_output()}
+{ToolUse("save", ["hello.py"], 'print("Hello world")', {"filepath": "hello.py", "content": 'print("Hello world")'}).to_output(tool_type)}
```

Comment:
Will this later be written like this?

```suggestion
{ToolUse("save", {"path": "hello.py", "content": 'print("Hello world")'}).to_output(tool_type)}
```

Or how can we eliminate the duplicated args?

Reply:
Fixed.